arch/arm/src/xmc4/xmc4_pwm.c

```diff
@@ -0,0 +1,1364 @@
+}
+
+/*****************************************************************************
+ * Name: xmc4_pwm_modifyreg32
+ *
+ * Description:
+ *   Modify a 32-bit register value by offset
+ *
+ *****************************************************************************/
+
+static void xmc4_pwm_modifyreg32(struct xmc4_pwm_s *priv,
+                                 uint32_t offset,
+                                 uint32_t clearbits,
+                                 uint32_t setbits)
+{
+  modifyreg32(priv->base + offset, clearbits, setbits);
+}
+
+/*****************************************************************************
+ * Name: pwm_set_period_match
+ *
+ * Description:
+ *   Set the period match register (CC4yPRS.PRS).
+ *   Must call pwm_shadow_transfert() after.
+ *
+ * Returned Value:
+ *   Zero on success; a negated errno value on failure
+ *
+ *****************************************************************************/
+
+static int pwm_set_period_match(struct xmc4_pwm_s *priv, uint16_t period_val)
+{
+  DEBUGASSERT(priv != NULL);
+
+  uint32_t cc4yprs_offset =
+      (uint32_t)(XMC4_CCU4_CC40PRS_OFFSET + 0x0100 * priv->slice);
+
+  xmc4_pwm_putreg32(priv, cc4yprs_offset, period_val);
+
+  pwminfo("PWM CCU4%d,CC4%d period is : %" PRId16 "\n",
+          priv->module, priv->slice, period_val);
+
+  return OK;
+}
+
+/*****************************************************************************
+ * Name: pwm_set_compare_match
+ *
+ * Description:
+ *   Set the comapre match register (CC4yCRS.CRS).
+ *   Must call pwm_shadow_transfert() after.
+ *
+ * Returned Value:
+ *   Zero on success; a negated errno value on failure
+ *
+ *****************************************************************************/
+
+static int pwm_set_compare_match(struct xmc4_pwm_s *priv,
+                                  uint16_t compare_val)
+{
+  DEBUGASSERT(priv != NULL);
+
+  uint32_t cc4ycrs_offset =
+      (uint32_t)(XMC4_CCU4_CC40CRS_OFFSET + 0x0100 * priv->slice);
+
+  xmc4_pwm_putreg32(priv, cc4ycrs_offset, compare_val);
+
+  pwminfo("PWM CCU4%d,CC4%d compare is : %" PRId16 "\n",
+          priv->module, priv->slice, compare_val);
+
+  return OK;
+}
+
+/*****************************************************************************
+ * Name: pwm_set_passive_level
+ *
+ * Description:
+ *   Set the passive level of the PWM slice.
+ *   Must call pwm_shadow_transfert() after.
+ *
+ * Returned Value:
+ *   Zero on success; a negated errno value on failure
+ *
+ *****************************************************************************/
+
+static int pwm_set_passive_level(struct xmc4_pwm_s *priv, uint8_t level)
+{
+  DEBUGASSERT(priv != NULL);
+
+  uint32_t passive_level = (uint32_t)(level == 1);
+
+  uint32_t cc4ypsl_offset =
+      (uint32_t)(XMC4_CCU4_CC40PSL_OFFSET + 0x0100 * priv->slice);
+
+  xmc4_pwm_putreg32(priv, cc4ypsl_offset, passive_level);
+
+  pwminfo("PWM CCU4%d,CC4%d passive level is : %" PRId8 "\n",
```

Comment:
No need of all the PRI stuff in nuttx. The difference between arch is already handled